### PR TITLE
New Cheatsheet: Python Datetime

### PR DESCRIPTION
**Pull request by danidee10**
## Description of new Instant Answer, or changes
This PR adds a new instant answer (Cheat sheet) for Pythons `DateTime` class, The most important part of this Cheat sheet is the Date formatting section.

## People to notify
@pjhampton The Python `DateTime` Cheatsheet is ready for review.

---

Instant Answer Page: https://duck.co/ia/view/python_datetime_cheatsheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@danidee10 Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
Thanks @danidee10, this is really taking shape! Now that you have cut through a cheatsheet, do you fancy working on a Fathead next?

**Comment by danidee10** [comment]
Yes i do, I've been reading the duckduckhack docs on the various fatheads, and i believe i'm ready to dive in

**Comment by GuiltyDolphin** [comment]
@danidee10 You might be able to cut back a bit on the date formatting section and focus on the library functions seeing as we have a cheat sheet for the formats already (https://duck.co/ia/view/unix_time_format_cheat_sheet) - you could probably change the ordering to emphasize the functions too.

**Comment by danidee10** [comment]
@GuiltyDolphin i didn't know a cheatsheet for unix time formats already existed. I'll let you guys know when i modify the cheatsheet.

File: share/goodie/cheat_sheets/json/python-datetime.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{
+    "id": "python_datetime_cheat_sheet",
+    "name": "Python Datetime",
+    "description": "python datetime and formatting",
+    "metadata": {
+        "sourceName": "OverAPI",
+        "sourceUrl": "http://overapi.com/python"
+    },
+    "aliases": ["date", "time"],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/python-datetime.json]: These aliases are too generic. So lets avoid them and add specific aliases

**Comment by danidee10** [comment]
@GuiltyDolphin and @jophab I've made the required changes. How does it look now?

Also i need some help, i couldn't figure out how to trigger the cheat-sheet with these queries (Without hard-coding it as an alias), and i think they're really relevant for the IA page to get more hits, because i search for things backward sometimes.

    datetime cheatsheet python
    datetime cheatsheet for python

but currently this query triggers it
    datetime python cheatsheet

**Comment by jophab** [comment]
@danidee10  This LGTM :+1: 

About `datetime cheatsheet python and
datetime cheatsheet for python` I dont think it is possible now. The regex used checks for triggering words at the end of query. Is that it? @GuiltyDolphin

**Comment by danidee10** [comment]
Seems this PR has stalled @GuiltyDolphin any opinions?

**Comment by pjhampton** [comment]
Hi @danidee10, 

I really appreciate your help with the Python side of things in the Programming mission. Cheat sheets are low-priority compared to Fatheads so reviews may take _a little longer_. 

You are yet to respond to / address the concerns in my review which I still believe to be valid. Would you mind responding?

**Comment by danidee10** [comment]
I've looked back and read the previous chats just to be sure. You only asked if i would love to work on a Fathead next (and i said Yes). after that i made some changes based on what @jophab and @GuiltyDolphin requested.

**Comment by pjhampton** [comment]
Hey @danidee10. I forgot to submit my review *facepalm*. You can see my responses now.

**Comment by danidee10** [comment]
@pjhampton I've made the requested changes. also would it be possible to trigger those queries i listed above?

**Comment by pjhampton** [comment]
Thanks @danidee10!

**Comment by daxtheduck** [comment]
@danidee10 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**